From 1d4bf810d964bfb3e34dc9a61b317d07d23c4fa4 Mon Sep 17 00:00:00 2001
From: Octavian Purdila <octavian.purdila@nxp.com>
Date: Wed, 8 Mar 2017 19:02:29 +0200
Subject: [PATCH 1555/5242] MLK-14388 mtd: gpmi: remove direct mtd->priv
 accesses

commit  cb3ad706411315d8942deadf8501c429d36da6c5 from
https://source.codeaurora.org/external/imx/linux-imx.git

mtd->priv is no longer pointing to the struct nand_chip it is attached
to.  Replace those accesses by mtd_to_nand() calls.

Signed-off-by: Octavian Purdila <octavian.purdila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
index 3c6354c..8afb1a4 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
@@ -367,7 +367,7 @@ static int set_geometry_for_large_oob(struct gpmi_nand_data *this)
 {
 	struct bch_geometry *geo = &this->bch_geometry;
 	struct mtd_info *mtd = &this->nand.mtd;
-	struct nand_chip *chip = mtd->priv;
+	struct nand_chip *chip = mtd_to_nand(mtd);
 	unsigned int block_mark_bit_offset;
 	unsigned int max_ecc;
 	unsigned int bbm_chunk;
-- 
1.7.9.5

